### PR TITLE
Add a linient Duration converter

### LINKED ISSUE #2452: Introduce lenient Duration converter
The Converter should accept values which start with a number by implicitly appending a `PT` to them.  If the value consists *only* of a number, it should implicitly treat the value as seconds.

### PR DESCRIPTION
**Pull request by machi1990**
Fixes #2452

### CODE REVIEW

File: core/runtime/src/main/java/io/quarkus/runtime/configuration/DurationConverter.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+     * The converter accepts a value which start with a number by implicitly appending `PT` to it.
+     * If the value consists only of a number, it implicitly treats the value as seconds.
+     * Otherwise, tries to convert the value assuming that it is in the accepted ISO-8601 duration format.
+     *
+     * @param value
+     * @return {@link Duration}
+     */
+    @Override
+    public Duration convert(String value) {
+        if (value.matches("[-+]?\\d+")) {
```
Comment (dmlloyd) [inline core/runtime/src/main/java/io/quarkus/runtime/configuration/DurationConverter.java]: Please precompile these into `Pattern` objects and store them in `static final` fields.

**Review by dmlloyd — CHANGES_REQUESTED** [review]
Just one minor change.

File: core/runtime/src/main/java/io/quarkus/runtime/configuration/DurationConverter.java (hunk 1 above)
Comment (machi1990) [reply core/runtime/src/main/java/io/quarkus/runtime/configuration/DurationConverter.java]: @dmlloyd thanks for the review. Addressing it now.

Comment (machi1990) [reply core/runtime/src/main/java/io/quarkus/runtime/configuration/DurationConverter.java]: Done.

**Review by aloubyansky — APPROVED** [review]

**Review by dmlloyd — APPROVED** [review]

**Pull request merged**